src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -3250,6 +3254,30 @@ internal EffectiveParameters(ImmutableArray<TypeWithAnnotations> types, Immutabl
             return paramRefKind;
         }
 
+        // In method group conversions, 'in' is allowed to match 'ref' and 'ref readonly' is allowed to match 'ref' or 'in'.
+        internal static bool AreRefsCompatibleForMethodConversion(RefKind x, RefKind y, CSharpCompilation compilation)
+        {
+            Debug.Assert(compilation is not null);
+            return x == y || IsRefMismatchAcceptableForMethodConversion(x, y, compilation);
+        }
+
+        // Only may be used to determine if warnings should be emitted.
+        internal static bool IsRefMismatchAcceptableForMethodConversion(RefKind x, RefKind y, CSharpCompilation compilationOpt)
+        {
+            if (x == RefKind.RefReadOnlyParameter)
+            {
+                return y is RefKind.Ref or RefKind.In;
+            }
+
+            if (y == RefKind.RefReadOnlyParameter)
+            {
+                return x is RefKind.Ref or RefKind.In;
+            }
+
+            return (x, y) is (RefKind.In, RefKind.Ref) or (RefKind.Ref, RefKind.In) &&
+                (compilationOpt is null || compilationOpt.IsFeatureEnabled(MessageID.IDS_FeatureRefReadonlyParameters));
```

Comment:
>compilationOpt is null [](http://example.com/codeflow?start=17&length=22)

This looks suspicious. Is there a scenario when we are trying to figure out an answer outside of context of a compilation?

Reply:
As the comment above this method says, it should be only used when reporting warnings, there the conversion is already successful, so no need to pass compilation.

Reply:
> As the comment above this method says, it should be only used when reporting warnings, there the conversion is already successful, so no need to pass compilation.

Ok, then why we are passing it in some cases and check the feature support?

Reply:
Perhaps we don't need to call this method from ```CheckRefReadonlyInMismatch``` at all? See my comment there.